### PR TITLE
Add javascript api for fast-math option

### PR DESCRIPTION
**Pull request by MaxGraey**
Relate to #3155

### CODE REVIEW

File: src/binaryen-c.cpp [hunk 1]
```diff
@@ -3570,6 +3570,10 @@ void BinaryenSetLowMemoryUnused(int on) {
   globalPassOptions.lowMemoryUnused = on != 0;
 }
 
+int BinaryenGetFastMath(void) { return globalPassOptions.fastMath; }
+
+void BinaryenSetFastMath(int value) { globalPassOptions.fastMath = value != 0; }
```
Comment (dcodeIO) [inline src/binaryen-c.cpp:3575]: These also need to be declared in `binaryen-c.h` using `BINARYEN_API`.

Comment (MaxGraey) [reply src/binaryen-c.cpp:3575]: right! Thanks

File: test/binaryen.js/fast-math.txt [hunk 2]
```diff
@@ -0,0 +1 @@
+// fastMath=true
```
Comment (dcodeIO) [inline test/binaryen.js/fast-math.txt:1]: Wouldn't printing the default result in `false` here?

Comment (MaxGraey) [reply test/binaryen.js/fast-math.txt]: it should. I just manually created this fixture artefact) Thanks again

File: src/binaryen-c.h [hunk 3]
```diff
@@ -2134,6 +2134,12 @@ BINARYEN_API int BinaryenGetLowMemoryUnused(void);
 // when optimizing. Applies to all modules, globally.
 BINARYEN_API void BinaryenSetLowMemoryUnused(int on);
 
+// Gets optimize floats without handling corner cases of NaNs.
```
Comment (dcodeIO) [inline src/binaryen-c.h:2137]: Perhaps:

> Gets whether fast math optimizations are enabled, ignoring for example corner cases of floating-point
math like NaN changes.

respectively on the setter:

> Enables or disables fast math optimizations, ignoring for example corner cases of floating-point math like NaN changes.

File: src/js/binaryen.js-post.js [hunk 4]
```diff
@@ -3046,6 +3046,16 @@ Module['setLowMemoryUnused'] = function(on) {
   Module['_BinaryenSetLowMemoryUnused'](on);
 };
 
+// Gets optimize floats without handling corner cases of NaNs.
```
Comment (dcodeIO) [inline src/js/binaryen.js-post.js:3049]: :)

**Review by dcodeIO — APPROVED** [review]
LGTM 👍

**Comment by MaxGraey** [comment]
Hardest pull request in my life 😂

**Pull request merged**